### ISSUE TITLE
Problem with Flyway with Username and Password

### ISSUE THREAD

**Issue by akefirad** [zonkyio/embedded-database-spring-test#58]
I'm trying to get the library working with Spring Boot + Postgres + Flyway. Everything works like a charm. But when I add a dedicated username/password for Flyway tests fail.  
I had a quick look at the [code](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration.java#L133)  and saw that Flyway Boot AutoConfiguration internally creates a data source if either `url` or `user` is not null. It seems this data source is not overridden to point to the embedded database. Am I missing something?

**Comment by tomix26** [comment]
It seems to be some Flyway's feature aimed to non-Spring applications. The internally created data source is not a spring bean, that is the reason why it's not being replaced by the embedded database.

I would recommend you to avoid this feature and use standard spring boot properties instead.
```
spring.datasource.url= # JDBC URL of the database.
spring.datasource.username= # Login username of the database.
spring.datasource.password= # Login password of the database.
```

Or, if you need a specific configuration only for Flyway, you can consider to have two separate data sources and use `@FlywayDataSource` annotation to specify which one is the Flyway's data source.

**Comment by akefirad** [comment]
That's right. It is indeed an instantiation using `new`. We have to have a different username for Flyway since the application user does not have DDL privileges. I didn't want to user `FlywayDataSource` since I use the default data source created by Spring. Now I have to create the two data source manullay :( But I guess there's no other way. Thanks.

**Comment by tomix26** [comment]
Another possible solution is to keep these problematic properties separated from integration tests. For example, you can use spring profiles for this purpose.

application.properties
```properties
spring.datasource.url=jdbc:postgresql://...
spring.datasource.username=user
spring.datasource.password=pass
```

application-prod.properties
```properties
spring.flyway.url=jdbc:postgresql://...
spring.flyway.user=flyway-user
spring.flyway.password=flyway-pass
```

The `prod` profile will be applied only to production, not to integration tests.

**Comment by akefirad** [comment]
Yeah, but we don't have profiles in production, not working for us :disappointed:
What I did though is to add an application initializer. I posted the code [here](https://stackoverflow.com/questions/57010132) for future reference. Thanks.

**Comment by tomix26** [comment]
Nice solution. Maybe I could implement it directly into the library. I'll think about it.
